packages/aws-cdk-lib/aws-codebuild/lib/project.ts

```diff
@@ -2240,6 +2241,112 @@ export class WindowsBuildImage implements IBuildImage {
+ *
+ * This class has a bunch of public constants that represent the most popular images.
+ *
+ * You can also specify a custom image using one of the static methods:
+ *
+ * - MacBuildImage.fromDockerRegistry(image[, { secretsManagerCredentials }])
+ * - MacBuildImage.fromEcrRepository(repo[, tag])
+ * - MacBuildImage.fromAsset(parent, id, props)
+ *
+ *
+ * @see https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-available.html
+ */
+export class MacBuildImage implements IBuildImage {
+  /**
+   * Corresponds to the standard CodeBuild image `aws/codebuild/macos-arm-base:14`.
+   */
+  public static readonly BASE_14: IBuildImage = new MacBuildImage({
+    imageId: 'aws/codebuild/macos-arm-base:14',
+    imagePullPrincipalType: ImagePullPrincipalType.CODEBUILD,
+  });
+
+  /**
+   * Makes an ARM MacOS build image from a Docker Hub image.
+   */
+  public static fromDockerRegistry(name: string, options: DockerImageOptions = {}): IBuildImage {
+    return new MacBuildImage({
+      ...options,
+      imageId: name,
+      imagePullPrincipalType: ImagePullPrincipalType.SERVICE_ROLE,
+    });
+  }
+
+  /**
+   * Makes an ARM MacOS build image from an ECR repository.
+   */
+  public static fromEcrRepository(repository: ecr.IRepository, tagOrDigest: string = 'latest'): IBuildImage {
+    return new MacBuildImage({
+      imageId: repository.repositoryUriForTagOrDigest(tagOrDigest),
+      imagePullPrincipalType: ImagePullPrincipalType.SERVICE_ROLE,
+      repository,
+    });
+  }
+
+  /**
+   * Uses an Docker image asset as a ARM MacOS build image.
+   */
+  public static fromAsset(scope: Construct, id: string, props: DockerImageAssetProps): IBuildImage {
+    const asset = new DockerImageAsset(scope, id, props);
+    return new MacBuildImage({
+      imageId: asset.imageUri,
+      imagePullPrincipalType: ImagePullPrincipalType.SERVICE_ROLE,
+      repository: asset.repository,
+    });
+  }
+
+  public readonly type = EnvironmentType.MAC_ARM as string;
+  public readonly defaultComputeType = ComputeType.MEDIUM;
+  public readonly imageId: string;
+  public readonly imagePullPrincipalType?: ImagePullPrincipalType;
+  public readonly secretsManagerCredentials?: secretsmanager.ISecret;
+  public readonly repository?: ecr.IRepository;
+
+  private constructor(props: MacBuildImageProps) {
+    this.imageId = props.imageId;
+    this.imagePullPrincipalType = props.imagePullPrincipalType;
+    this.secretsManagerCredentials = props.secretsManagerCredentials;
+    this.repository = props.repository;
+  }
+
+  public validate(buildEnvironment: BuildEnvironment): string[] {
+    const errors: string[] = [];
+
+    if (buildEnvironment.computeType && isLambdaComputeType(buildEnvironment.computeType)) {
+      errors.push('Mac images do not support Lambda compute types');
+    }
+
+    if (buildEnvironment.computeType != ComputeType.MEDIUM) {
+      errors.push('Mac images only support medium compute type');
+    }
```

Comment:
@alanxluc .. I think this type of validation that there is a big probability to change, and I think it should not be part of CDK validations.

Reply:
Ack, will revise. Is the other validation good to keep?

Reply:
yes, they look reasonable. Thanks @alanxluc for your contribution :)